vulnerabilities/import_runner.py

```diff
@@ -58,6 +61,46 @@ class ImportRunner:
     def __init__(self, importer: models.Importer, batch_size: int):
         self.importer = importer
         self.batch_size = batch_size
+        self._bulk_create_vuln_pkg_refs = []
+        self._bulk_update_vuln_pkg_refs = []
+        self._bulk_create_vuln_refs = []
+    
+    def __del__(self):
+        models.Vulnerability_Package_Relation.objects.bulk_create(self._bulk_create_vuln_pkg_refs, ignore_conflicts=True)
+        models.Vulnerability_Package_Relation.objects.bulk_update(self._bulk_update_vuln_pkg_refs,['is_vulnerable'])
+        models.VulnerabilityReference.objects.bulk_create(self._bulk_create_vuln_refs, ignore_conflicts=True)
+
+    # TODO: Currently the implementation may contain duplicate model instances in `_bulk_create_vuln_refs` and `_bulk_create_vuln_pkg_refs`
+    #       to make this work , `ignore_conflicts=True` is used. Find some way to remove duplicates of model instances(before writing them db)
+    #       and remove the   `ignore_conflicts=True` flag.
+    @property  
+    def bulk_create_vuln_pkg_refs(self):
```

Comment:
Property getters should not have side-effects. IMHO this approach has too much magic, including the need for the `__del__()` method.

Reply:
Agreed this  and `ignore_conflicts=True` are totally bad practices. Good news is after tinkering about this, I can totally get rid of these magic methods and properties, even maintain `ImporterRunner`'s API like the old one.